packages/ckeditor5-dev-ci/bin/circle-workflow-notifier.js

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const { execSync } = require( 'child_process' );
+const fetch = require( 'node-fetch' );
+const minimist = require( 'minimist' );
+const processJobStatuses = require( '../lib/process-job-statuses' );
+
+// This script allows the creation of a new job within a workflow that will be executed
+// in the end, when all other jobs will be finished or errored.
+//
+// Below, you can find an example workflow.
+//
+//   ┌─────┐     ┌─────┐
+//   │Job A├────►│Job B├──────────┐
+//   └─────┘     └─────┘          ▼
+//                           ┌────────┐
+//                           │Notifier│
+//                           └────────┘
+//   ┌─────┐                      ▲
+//   │Job C├──────────────────────┘
+//   └─────┘
+//
+// Job A triggers Job B, and Job C has no dependencies. When all jobs are done, we would like to execute Notifier.
+//
+// The Notifier job should also be executed when Job A ends with an error. In such a case, Job B is still blocked.
+// Hence, we need to iterate over all jobs and verify if their dependencies ended with an error to unlock
+// executing the final part of the workflow.
+
+const {
+	/**
+	 * Required. CircleCI API token used for obtaining data about the build from API.
+	 */
+	CKE5_CIRCLE_TOKEN,
+
+	// Variables that are available by default in Circle environment.
+	CIRCLE_WORKFLOW_ID,
+	CIRCLE_JOB
+} = process.env;
+
+const { task } = parseArguments( process.argv.slice( 2 ) );
+const FAILING_PARENT_FLAG = 'failing_parent';
+
+waitForOtherJobsAndSendNotification()
+	.catch( err => {
+		console.error( err );
+
+		process.exit( 1 );
+	} );
+
+async function waitForOtherJobsAndSendNotification() {
+	const jobs = await getOtherJobsData();
+
+	processJobStatuses( jobs, FAILING_PARENT_FLAG );
```

Comment:
Missing assignment of the return value from `processJobStatuses()`, because this function operates on a cloned data.
```suggestion
	const jobs = processJobStatuses( await getOtherJobsData(), FAILING_PARENT_FLAG );
```